### ISSUE TITLE
Remove node_a from the joints

### ISSUE THREAD

**Issue by ettiSurreal** [godotengine/godot-proposals#5778]
### Describe the project you are working on

Testing features

### Describe the problem or limitation you are having in your project

`Node A` in 2D and 3D joints is a seemingly redundant property, that only makes it less immediately clear of how they work, and makes them unnecesarily more difficult to set up and add through code.

For example in the case of a simple grapple hook, you'll have to create the joint outside of its users body (otherwise it might sometimes lead to weird behavior) and specify that you want it connected to it. Both of these steps can be avoided if it was just possible to spawn the joint as a child of the body.
The behavior is also not clear to new users, who might think the start of the link is the position of the first body.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Remove the `Node A` property from the joint class, the start of the link is now entirely dependent on the position of the joint node. In order to bind two bodies together you now simply need to parent the joint node to the first body.

This solves the issues i've mentioned before, at seemingly no loss in functionality.

My only guess to why joints work like this in the first place is the `exclude_nodes_from_collision` property, which might be replicatable by excluding the collision between the joints immediate parent and the connected body, a similar behavior can be seen in the raycast node.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

`Node A` is removed from Joint2D and 3D, you now bind the first body by directly parenting the joint to it in the scene tree.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Doesn't seem like it.

### Is there a reason why this should be core and not an add-on in the asset library?

Changing a built in class.

**Comment by Mickeon** [comment]
I'm going to assume the better judgement of this user and by extension supporting removing an useless property. However, because I have not used **Joint2D** and **Joint3D** extensively, and perhaps speaking for others, I cannot say for sure the alternative is a good idea.

**Comment by YuriSizov** [comment]
> Doesn't seem like it.

You can have a script that you attach to your every joint node which assigns to the node A property joint's parent node, using `get_parent()` (maybe in a combination with `get_path()`, if the property takes a NodePath instead of a reference).

Removing the property entirely seems to remove the possibility to have this node in a separate branch. Not saying this is a valid need, but that's definitely a loss of functionality. We could also add a property instead, to preserve the existing functionality but streamline using the hierarchy instead. Can be a boolean flag that makes the joint to automatically take its parent and disables editing the node A property.

**Comment by ettiSurreal** [comment]
> Removing the property entirely seems to remove the possibility to have this node in a separate branch. Not saying this is a valid need, but that's definitely a loss of functionality.

I cannot think of any use cases where this would be used instead of just parenting the joint to the body, and the functionality can likely be replicated by simply making a plain node that you can assign the joints path to.
And sometimes is it worth having functionality for hypothetical cases that makes every other use case harder?
Boolean flag could work though, and if the original implementation would be severely underused in favor of it, it can be eventually removed in a compat breaking release.

**Comment by YuriSizov** [comment]
> And sometimes is it worth having functionality for hypothetical cases that makes every other use case harder?

No, we prefer not to have functionality for hypothetical cases. Which in turn makes me think this is intentional and there is a good reason for this.
